File: Java/programas/Sobrecarga Perímetro figuras/src/Triangulo.java
public class Triangulo {
     //Atributos privados.
    private float perimetro1, perimetro2, perimetro3;

    //Funcinoes sobre cargadas con distinta cantidad de parámetros.
    //Regresa el perímetro de los tipos de triángulos en una variable.
    public float Perimetro(float lado){
        return perimetro1 = lado * 3;
    }

    public float Perimetro(float altura, float Base){
        return perimetro2 = (altura + Base) * 2;
    }

    public float Perimetro(float lado1, float lado2, float lado3){
        return perimetro3 = lado1 + lado2 + lado3;
    }

    //Procedimiento que despliega los perímetros.
    public void Resultados(){
        System.out.println("\n---Resultados---\n");
        System.out.println("Perímetro del triángulo equilatero: " + perimetro1 + "u.");
        System.out.println("Perímetro del triángulo isóseles: " + perimetro2 + "u.");
        System.out.println("Perímetro del triángulo escaleno: " + perimetro3 + "u.");
    }
}

File: C++/programas/P-Sobrecarga de metodos/P-Sobrecarga de metodos/Triangulo.cpp
#include <iostream> //Cabecera para entrada y salida de datos.
#include "Triangulo.h" //Incluye cabecera de la clase.

//Funci?n sobrecargada.
float Triangulo::Perimetro(float l)
{
    //Regresa el per?metro de un tri?ngulo equil?tero.
    return l * 3;
}

float Triangulo::Perimetro(float base, float altura)
{
    //Regresa el per?metro de un tri?ngulo is?seles.
    return (altura * 2) + base;
}

float Triangulo::Perimetro(float l1, float l2, float l3)
{
    //Regresa el per?metro de un tri?ngulo escaleno.
    return l1 + l2 + l3;
}

//Procedmiento que despliega los resultados obtenidos.
void Triangulo::Resultados(float p1, float p2, float p3)
{
    //Despliega resultado de calculos de per?metros.
    std::cout << "\n--Resultados--\n\n";
    std::cout << "Per?metro del tri?ngulo equil?tero: " << p1 << "u.\n";
    std::cout << "Per?metro del tri?ngulo is?seles: " << p2 << "u.\n";
    std::cout << "Per?metro del tri?ngulo escaleno: " << p3 << "u.";
}

Triangulo::~Triangulo()
{
    //Despliega mensaje de objeto destruido.
    std::cout << "\n\nObjeto Triangulo destruido.\n";
}
